### PR TITLE
Gh-3113: Fix regex in create release branch workflow

### LINKED ISSUE #3113: Fix create release branch workflow regex
In the "Bump latest tag variable version" step, there is an incorrect regex:
https://github.com/gchq/Gaffer/blob/1ef3621323064ab6a15c84fa554d35fe1dea003d/.github/workflows/create-release-branch.yaml#L41
It should set the patch version to 0 but does not

### PR DESCRIPTION
**Pull request by t92549**
# Related issue

- Resolve #3113

### CODE REVIEW

File: .github/workflows/create-release-branch.yaml [hunk 1]
```diff
@@ -38,7 +38,7 @@ jobs:
 
     - name: Bump latest tag variable version
       if: ${{ !github.event.inputs.version }}
-      run: echo "RELEASE_VERSION=$(echo ${{ env.CURRENT_VERSION }} | sed -r 's/([0-9]+)\.([0-9]+)\.([0-9]+)/echo \1.$((\2+1)).\3/' | sh)" >> $GITHUB_ENV
+      run: echo "RELEASE_VERSION=$(echo ${{ env.CURRENT_VERSION }} | sed -r 's/([0-9]+)\.([0-9]+)\.([0-9]+)/echo \1.$((\2+1)).0/' | sh)" >> $GITHUB_ENV
```
Comment (tb06904) [inline .github/workflows/create-release-branch.yaml:41]: Just a thought a simple `awk` script might be a better choice here e.g. something like this should do the job
```bash
cho "RELEASE_VERSION=$(echo ${{ env.CURRENT_VERSION }} | awk -F. -v OFS=. '{$2 += 1 ; $3 = 0; print}')" >> $GITHUB_ENV
```

**Review by GCHQDeveloper314 — APPROVED** [review]

File: .github/workflows/create-release-branch.yaml (hunk 1 above)
Comment (t92549) [reply .github/workflows/create-release-branch.yaml:41]: @tb06904 if this sounds good to you, I think for now I will leave this as it isn't crucial to solve the issue and this regex is present on the other repos. Worth maybe thinking about improving though when we come to refactor the release pipeline generally.

Comment (GCHQDeveloper314) [reply .github/workflows/create-release-branch.yaml:41]: I would echo this. There are at least as many instances of this `sed` command as there are Gaffer repos, probably a couple more. A change like this should be done as a refactor to keep consistency.

Comment (tb06904) [reply .github/workflows/create-release-branch.yaml:41]: That's fair doe we have a ticket for the refactor?

**Review by tb06904 — APPROVED** [review]

File: .github/workflows/create-release-branch.yaml (hunk 1 above)
Comment (t92549) [reply .github/workflows/create-release-branch.yaml:41]: Not yet, we should probably make one though. I think the main points I would like to do are:
- removing the `ADMIN_GITHUB_TOKEN`
- removing release branches

**Pull request merged**